### PR TITLE
perf: Add member id to bundle request model

### PR DESCRIPTION
**Pull request by levansuper**

### CODE REVIEW

File: src/awala.spec.ts [hunk 1]
```diff
@@ -34,206 +44,243 @@ describe('awala', () => {
...
 
-      await createMemberBundleRequest(methodInput, serviceOptions);
+    const countResult = await memberBundleRequestModel.count({
+      publicKeyId: memberPublicKey._id.toString(),
+    });
+    expect(countResult).toBe(1);
+  });
 
-      const countResult = await memberBundleRequestModel.count({
+  test('Invalid public key id should be refused', async () => {
```
Comment (gnarea) [inline src/awala.spec.ts:125]: "invalid" is too ambiguous.

```suggestion
  test('Unknown public key id should be refused', async () => {
```

File: src/awala.spec.ts [hunk 2]
```diff
@@ -34,206 +44,243 @@ describe('awala', () => {
...
-      });
-      expect(countResult).toBe(1);
+      },
+      serviceOptions,
+    );
+
+    expect(result.didSucceed).not.toBeTrue();
+  });
+
+  test('Invalid public key Id should not create member bundle request', async () => {
```
Comment (gnarea) [inline src/awala.spec.ts:137]: Looking at the other tests in this file, wouldn't it be more consistent to bundle this together with the test above? See:

https://github.com/relaycorp/veraid-authority/blob/291f81f7bbf8a4265341290a8e800642aba5a38c/src/awala.spec.ts#L52

File: src/awala.ts [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
   options: ServiceOptions,
-): Promise<SuccessfulResult<undefined>> {
+): Promise<Result<undefined, undefined>> {
   const memberBundleRequestModel = getModelForClass(MemberBundleRequestModelSchema, {
     existingConnection: options.dbConnection,
   });
+  const memberPublicKeyModel = getModelForClass(MemberPublicKeyModelSchema, {
+    existingConnection: options.dbConnection,
+  });
+  const publicKey = await memberPublicKeyModel.findById(requestData.publicKeyId);
```
Comment (gnarea) [inline src/awala.ts:22]: We don't need the record, right? Let's do an _exists_ query

**Review by gnarea — CHANGES_REQUESTED** [review]

File: src/awala.ts (hunk 3 above)
Comment (levansuper) [reply src/awala.ts:22]: actually, we need this to get the member Id
https://github.com/relaycorp/veraid-authority/pull/136/files#diff-725e18d8b08af438b9930029c9d9f1d74127beb38f17a39bc726315266bd435cR40

File: src/awala.spec.ts (hunk 1 above)
Comment (gnarea) [reply src/awala.spec.ts]: That works too

**Review by gnarea — APPROVED** [review]
Thanks!

**Pull request merged**